### PR TITLE
Implements the `noop` cluster manager and implements `List` in tanzu package

### PR DESCRIPTION
**Pull request by jpmcb**
## What this PR does / why we need it

All changes related to the standalone cluster overhaul

- Enables users to bring their own cluster with their own kubeconfig
- Removes listing as part of the cluster package and moves it's functionality to the tanzu package. Now, users can list all clusters under the config dir (wheather they were created by our provider or were `provider: none`)
- Adds error logging in kapp controller manager
- Updates calls to NewKindClusterManager to be generic

### CODE REVIEW

**Review by joshrosso — CHANGES_REQUESTED** [review]
Nothing wrong with this PR just want to be included in the final review before merge.

**Review by stmcginnis — CHANGES_REQUESTED** [review]
Looks great! One minor thing I'd like to see changed, but otherwise ready to go.

**Review by stmcginnis — APPROVED** [review]
Thanks! I'll leave to @joshrosso to push the big button.

**Review by joshrosso — COMMENTED** [review]
Our Tanzu API needs to support bringing the `kubeconfig` in as a byte array rather than a path to a file. This will be more common for those that call us programatically.

**Review by joshrosso — CHANGES_REQUESTED** [review]
overall looks great, added some small recommendations for change.

**Pull request merged**
